### PR TITLE
COMMERCE-11867 Add functional tests for OOTB Account Role of Supplier to apply discounts to their own products

### PR DESCRIPTION
**Pull request by jaydawu**
Test plan: https://liferay.atlassian.net/browse/COMMERCE-11867

### CODE REVIEW

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase [hunk 1]
```diff
@@ -655,4 +655,189 @@ definition {
...
+			CommerceAcceleratorsInitializer.changeProductsSorting(sortingType = "Name Ascending");
+
+			CommerceAcceleratorsInitializer.gotoMiniumProductDetailsPage(productName = "Brake Pads");
+
+			CommerceAcceleratorsInitializer.viewDiscountedProductDetailsPage(
+				discountPercentage = 20,
+				productDiscountedPrice = "15.12",
+				productListPrice = "21.00",
+				productPromoPrice = "18.90");
+		}
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase:840]: Save as above, please use a buyer user instead

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase [hunk 2]
```diff
@@ -655,4 +655,189 @@ definition {
...
+			CommerceAcceleratorsInitializer.changeProductsSorting(sortingType = "Name Ascending");
+
+			CommerceAcceleratorsInitializer.gotoMiniumProductDetailsPage(productName = "Brake Pads");
+
+			CommerceAcceleratorsInitializer.viewDiscountedProductDetailsPage(
+				discountPercentage = 10,
+				productDiscountedPrice = "17.01",
+				productListPrice = "21.00",
+				productPromoPrice = "18.90");
+		}
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase:759]: The supplier shouldn't be the account to check the store front. Use a business account with a buyer user instead.

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase [hunk 3]
```diff
@@ -655,4 +655,189 @@ definition {
...
+			Check.checkToggleSwitch(
+				key_toggleSwitchLabel = "Override Discount",
+				locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
+
+			Type(
+				key_text = "L1",
+				locator1 = "TextInput#ANY",
+				value1 = 20);
+
+			Button.clickSave();
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase:787]: Check if this part can be included in an existing macro.

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase [hunk 4]
```diff
@@ -655,4 +655,189 @@ definition {
...
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase:814]: Check if this part can be included in an existing macro.

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase [hunk 5]
```diff
@@ -655,4 +655,189 @@ definition {
...
+		}
+
+		task ("Then the discounts of the products are updated by supplier user accordingly") {
+			Navigator.openWithAppendToBaseURL(urlAppend = "web/minium/catalog");
+
+			CommerceAcceleratorsInitializer.changeProductsSorting(sortingType = "Name Ascending");
+
+			CommerceAcceleratorsInitializer.gotoMiniumProductDetailsPage(productName = "ABS Sensor");
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase:822]: You can navigate directly to the product detail page
```suggestion
			Navigator.openWithAppendToBaseURL(urlAppend = "web/minium/p/abs-sensor");
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase [hunk 6]
```diff
@@ -655,4 +655,189 @@ definition {
...
+			CommerceAcceleratorsInitializer.viewDiscountedProductDetailsPage(
+				discountPercentage = 20,
+				productDiscountedPrice = "40.00",
+				productListPrice = "50.00");
+
+			Navigator.openWithAppendToBaseURL(urlAppend = "web/minium/catalog");
+
+			CommerceAcceleratorsInitializer.changeProductsSorting(sortingType = "Name Ascending");
+
+			CommerceAcceleratorsInitializer.gotoMiniumProductDetailsPage(productName = "Brake Pads");
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase:833]: Same as above

**Review by lorenzolapescara — CHANGES_REQUESTED** [review]

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:stop

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:stop

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase [hunk 7]
```diff
@@ -655,4 +655,178 @@ definition {
...
+
+		task ("And Supplier account is linked to Minium catalog") {
+			var accountEntryId = JSONAccountEntryAPI._getAccountEntryId(accountEntryName = "Commerce Supplier");
+
+			CommerceJSONCatalogsAPI._patchCommerceCatalog(
+				accountId = ${accountEntryId},
+				catalogName = "Minium");
+		}
+
+		task ("When one discount for the price product 'ABS Sensor' and one for the promotion product 'Brake Pads' are created by admin") {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase:717]:
```suggestion
		task ("When the admin creates a discount for products with base price and promo price") {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase [hunk 8]
```diff
@@ -655,4 +655,178 @@ definition {
...
+			AssertTextEquals(
+				locator1 = "Message#EMPTY_STATE_INFO",
+				value1 = "No Results Found");
+
+			CommerceProducts.searchAndGoToProductEntry(entryProductName = "MIN93018");
+
+			SelectFrame.selectFrameNoLoading(locator1 = "CommerceEntry#IFRAME_SIDE_PANEL");
+
+			CommercePromotions.setOverrideDiscount(overrideDiscounts = "L1-20");
+		}
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase:805]: Make these 2 separate tasks

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase [hunk 9]
```diff
@@ -655,4 +655,178 @@ definition {
...
+
+			ApplicationsMenu.gotoPanel(panel = "Commerce");
+
+			AssertElementNotPresent(
+				key_category = "Pricing",
+				key_portlet = "Discounts",
+				locator1 = "ApplicationsMenu#PORTLET");
+		}
+
+		task ("When the supplier user manages the products that have discounts to other discounts") {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase:769]: Apply a similar description for the promotion price task
```suggestion
		task ("When the supplier user adds a new price entry with discount set in a new price list") {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase [hunk 10]
```diff
@@ -655,4 +655,178 @@ definition {
 		}
 	}
 
+	@description = "This is a test for COMMERCE-11868. As a supplier account user, I can manage discounts associated with my products via Price Lists and Promotions Panels"
+	@priority = 5
+	test CanSupplierUserManageProductsWithDiscountsLinkedToCatalog {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase:660]:
```suggestion
	test CanSupplierUserManageDiscountsInPriceListsAndPromotions {
```

**Review by lorenzolapescara — CHANGES_REQUESTED** [review]
@jaydawu just a couple changes more, thank you!

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Review by lorenzolapescara — APPROVED** [review]

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:forward

**Comment by jaydawu** [comment]
ci:forward

**Comment by jaydawu** [comment]
Hi @victoroscuro please help monitor this, no relevant errors to this pull, it's good enough to be forwarded, but just has some errors when forwarding, needs to be retried, thanks!

**Comment by victoroscuro** [comment]
ci:forward

**Pull request closed without merging**